pkg/scale/decode.go

```diff
@@ -175,76 +221,88 @@ func (ds *decodeState) decodeCustomPrimitive(dstv reflect.Value) (err error) {
 		temp = reflect.New(reflect.TypeOf(false))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Bool: %w", err)
 			break
 		}
 	case reflect.Int:
 		temp = reflect.New(reflect.TypeOf(int(1)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Int: %w", err)
 			break
 		}
 	case reflect.Int8:
 		temp = reflect.New(reflect.TypeOf(int8(1)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Int8: %w", err)
 			break
 		}
 	case reflect.Int16:
 		temp = reflect.New(reflect.TypeOf(int16(1)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Int16: %w", err)
 			break
 		}
 	case reflect.Int32:
 		temp = reflect.New(reflect.TypeOf(int32(1)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Int32: %w", err)
 			break
 		}
 	case reflect.Int64:
 		temp = reflect.New(reflect.TypeOf(int64(1)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Int64: %w", err)
 			break
 		}
 	case reflect.String:
 		temp = reflect.New(reflect.TypeOf(""))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling String: %w", err)
 			break
 		}
 	case reflect.Uint:
 		temp = reflect.New(reflect.TypeOf(uint(0)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Uint: %w", err)
 			break
 		}
 	case reflect.Uint8:
 		temp = reflect.New(reflect.TypeOf(uint8(0)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Uint8: %w", err)
 			break
 		}
 	case reflect.Uint16:
 		temp = reflect.New(reflect.TypeOf(uint16(0)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Uint16: %w", err)
 			break
 		}
 	case reflect.Uint32:
 		temp = reflect.New(reflect.TypeOf(uint32(0)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Uint32: %w", err)
 			break
 		}
 	case reflect.Uint64:
 		temp = reflect.New(reflect.TypeOf(uint64(0)))
 		err = ds.unmarshal(temp.Elem())
 		if err != nil {
+			err = fmt.Errorf("unmarshaling Uint64: %w", err)
 			break
 		}
 	default:
-		err = fmt.Errorf("unsupported type for custom primitive: %T", in)
+		err = fmt.Errorf("%w: %T", errUnsupportedType, in)
```

Comment:
You should export `errUnsupportedType` since it's (most likely) wrapped up and escapes the scale package.
Unless it's logged in the current package, or handled in the current package with `errors.Is` for example, but I don't think that's the case.